schnorr_pubkey.go

```diff
@@ -0,0 +1,128 @@
+package secp256k1
+
+// #include "./depend/secp256k1/include/secp256k1_extrakeys.h"
+// #include "./depend/secp256k1/include/secp256k1_schnorrsig.h"
+import "C"
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/pkg/errors"
+)
+
+const (
+	// SerializedSchnorrPublicKeySize defines the length in bytes of a SerializedSchnorrPublicKey
+	SerializedSchnorrPublicKeySize = 32
+)
+
+// SchnorrPublicKey is a PublicKey type used to sign and verify Schnorr signatures.
+// The struct itself is an opaque data type that should only be created via the supplied methods.
+type SchnorrPublicKey struct {
+	pubkey C.secp256k1_xonly_pubkey
+}
+
+// SerializedSchnorrPublicKey is a is a byte array representing the storage representation of a compressed or uncompressed SchnorrPublicKey
+type SerializedSchnorrPublicKey [SerializedSchnorrPublicKeySize]byte
+
+// IsEqual returns true if target is the same as key.
+func (key *SchnorrPublicKey) IsEqual(target *SchnorrPublicKey) bool {
+	if key == nil && target == nil {
+		return true
+	}
+	if key == nil || target == nil {
+		return false
+	}
+	serializedKey, err1 := key.Serialize()
+	serializedTarget, err2 := target.Serialize()
+
+	if err1 != nil && err2 != nil { // They're both zeroed, shouldn't happen if a constructor is used.
```

Comment:
Don't assume a function can return only one error, instead, make an errZeroed and compare to it

Reply:
That's fair, will do